apps/jobs/views.py

```diff
@@ -1260,9 +1248,37 @@ def update_submission(request, challenge_pk):
 
     if request.method == "PATCH":
         submission_pk = request.data.get("submission")
+        submission = get_submission_model(submission_pk)
+        if (
+            request.FILES.get("submission_input_file")
```

Comment:
Also, it would be good if you add a `if else` here. Update the `submission_input_file` if the challenge is static code upload challenge. Else if the challenge is code upload challenge do the other update. Check for the `request.FILES.get("submission_input_file")` should be inside if.

Reply:
the current flow is
if File_submitted + static_code_upload_challenge: Do our thing and return
"Prev flow"

Are you suggesting
if static_code_upload_challenge:
    if File_submitted: Do our thing
else:
     "Prev flow"

The reason for not following the second design is that we are using the Prev flow as well for the static_code_upload_challenge.

Reply:
I see, then above changes are fine